includes/Core/Authentication/Clients/OAuth_Client_Base.php

```diff
@@ -0,0 +1,348 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Authentication\Clients\OAuth_Client_Base
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Authentication\Clients;
+
+use Exception;
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Authentication\Credentials;
+use Google\Site_Kit\Core\Authentication\Exception\Google_Proxy_Code_Exception;
+use Google\Site_Kit\Core\Authentication\Google_Proxy;
+use Google\Site_Kit\Core\Authentication\Profile;
+use Google\Site_Kit\Core\Authentication\Token;
+use Google\Site_Kit\Core\Permissions\Permissions;
+use Google\Site_Kit\Core\Storage\Encrypted_Options;
+use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Core\Storage\User_Options;
+
+/**
+ * Base class for connecting to Google APIs via OAuth.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+abstract class OAuth_Client_Base {
+
+	const OPTION_ACCESS_TOKEN            = 'googlesitekit_access_token';
+	const OPTION_ACCESS_TOKEN_EXPIRES_IN = 'googlesitekit_access_token_expires_in';
+	const OPTION_ACCESS_TOKEN_CREATED    = 'googlesitekit_access_token_created_at';
+	const OPTION_REFRESH_TOKEN           = 'googlesitekit_refresh_token';
```

Comment:
Yeah I left them there for BC. Not super critical, but I also don't think we really should move them to the new `Token` class since we eventually want to consolidate the options anyway. I'd prefer to therefore just leave them as is at this point. The BC considerations are small, but also moving them wouldn't be that beneficial.